test/StringPrefixTest.php

```diff
@@ -0,0 +1,76 @@
+<?php
+/**
+ * Zend Framework (http://framework.zend.com/)
+ *
+ * @link      http://github.com/zendframework/zf2 for the canonical source repository
+ * @copyright Copyright (c) 2005-2015 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license   http://framework.zend.com/license/new-bsd New BSD License
+ */
+
+namespace ZendTest\Filter;
+
+use PHPUnit\Framework\TestCase;
+use Zend\Filter\Exception\InvalidArgumentException;
+use Zend\Filter\StringPrefix as StringPrefixFilter;
+
+class StringPrefixTest extends TestCase
+{
+    /**
+     * @var StringPrefixFilter
+     */
+    protected $filter;
+
+    /**
+     * @return void
+     */
+    public function setUp()
+    {
+        $this->filter = new StringPrefixFilter();
+    }
+
+    /**
+     * Ensures that the filter follows expected behavior
+     *
+     * @return void
+     */
+    public function testBasic()
+    {
+        $filter = $this->filter;
+
+        $prefix = 'ABC123';
+        $filter->setPrefix($prefix);
+
+        $this->assertStringStartsWith($prefix, $filter('sample'));
```

Comment:
Assert methods are static so calling them with `$this->` is not correct.

Reply:
@waltertamboer I'll fix this next week, because I'm on holiday.

Reply:
@waltertamboer @thexpand 
No change is needed here. `$this->` is basically used everywhere in tests.

Reply:
Background: https://discourse.zendframework.com/t/this-assert-vs-self-assert/448

Reply:
Read it. I still don't get why we would promote bad syntax. It's not the standard I would expect from Zend Framework.

Reply:
> It's not the standard I would expect from Zend Framework.

"There is no right way": https://phpunit.readthedocs.io/en/7.3/assertions.html#static-vs-non-static-usage-of-assertion-methods

Reply:
I don't agree. There is a right way. The paragraphs in the phpunit docs are just extremely awkward to read.

> The short answer is: there is no right way. And there is no wrong way, either. It is a matter of personal preference.

Which can be rephrased as: "it doesn't matter". No, it's not a matter of personal preference. It's a matter of writing semantically correct code. Basically the arguments @Ocramius used in the discussion on Discourse.

But it's ok, I've made my argument and it won't change the fact that ZF is currently using `$this`. So let's move on.

Thanks for your time and patience.

Reply:
@waltertamboer 
I promote nothing here, it is a purely pragmatic decision: It is a code review and not a coding standard or semantically correct code discussion.

Please do not get me wrong, your comments are important, but this is the wrong place and after the merge of this pull request your comments are lost.

(_end of offtopic_)